### DISCUSSION TITLE
(1*0.4)(1*2)(1*0.2) is valid or invalid?

### DISCUSSION THREAD

**Discussion by inapeace0** [josdejong/mathjs#2828] [Q&A]
I doubt that this expression is valid with math.js module.

`(1*0.4)(1*2)(1*0.2)`

Actually, the original expression is `(1*0.4)+(1*2)+(1*0.2)`, and of course, it's valid with this module.

**Comment by gwhitney** [comment]
Well, if you type that exact expression into the "demo" at mathjs.org, you will see that it responds with a value of 0.16 and no indications of any problem or error. So indeed, it is a valid expression.

**Reply by inapeace0** [reply]
Yes, I see. With math.js, it's detected it's valid.
`(1*0.4)(1*2)(1*0.2) = (1*0.4)*(1*2)*(1*0.2) = 0.16`
But why I asked it is that it must be detected as invalid one in my project. In other words, there must be operators(+, -, *, /) between two operators ')' and '('.

**Reply by gwhitney** [reply]
As far as I know, there is not currently any mechanism in mathjs to disallow implicit multiplication, i.e. require an explicit `*` symbol to multiply juxtaposed subexpressions. The `parse()` function does have an "options" argument, so if you need it, you could submit a PR implementing a "multiplication" option with values 'allowImplicit' or 'onlyExplicit' (say) to control this. But yes, as far as I know it would currently require a code change to disallow implicit multiplication in expressions.

**Reply by josdejong** [reply]
If you want to disable implicit multiplication right now, you can do the following: parse the expression with `math.parse`, and then traverse the created expression tree to see if there is an occurrence of an `OperatorNode` which has a property `implicit: true`, and if so, throw an error. If all is ok you can evaluate the parsed expression.

Docs: https://mathjs.org/docs/expressions/expression_trees.html
